packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts

```diff
@@ -33,13 +34,39 @@ export class Scope {
   }
 
   public fullName(attributeName: string) {
+    return `${this.baseName}.${attributeName}`;
+  }
+
+  public get baseName() {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     let scope: Scope | undefined = this;
-    let name = attributeName;
+    let name = "";
     while (scope) {
-      name = `${scope.name}.${name}`;
+      if (name === "") {
+        name = scope.name;
+      } else {
+        name = `${scope.name}.${name}`;
+      }
       scope = scope.parent;
     }
     return name;
   }
+
+  public get attributePath() {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    let scope: Scope | undefined = this;
+    const path: string[] = [];
+    while (scope) {
+      path.unshift(scope.name);
+      scope = scope.parent;
+    }
+
+    return path.map((p, i) => {
+      if (i === 0) {
+        return p;
+      }
+      const last = path[i - 1];
+      return p.replace(`${last}_`, "");
```

Comment:
I feel like previous fits better 😄

```suggestion
      const prev = path[i - 1];
      return p.replace(`${prev}_`, "");
```